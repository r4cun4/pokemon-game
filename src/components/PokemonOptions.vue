<template>
  <div class="options-container">

    <ul>
        <li
          v-for="pokemon in pokemons"
          :key="pokemon.id"
          @click="$emit( 'selectedPokemon', pokemon.id )"
          @mouseover="speechNames"
        >
          {{ pokemon.name }}
        </li>
    </ul>

<!-- colocar el :key es una buena practica para identificar el elemento de manera unica -->

  </div>
</template>

<script>
export default {
// Se definen las props que se pasaron como argumento en la PokemonPage en el llamado del componente
    props: {
        pokemons: {
            type: Array,
            required: true
        }
    },
    methods: {
        speechNames(e) {

            const msg = new SpeechSynthesisUtterance()
            msg.text = ''
            msg.lang = 'es-MX'

            if(e.target.matches('li:nth-child(1)')) {
                msg.text = this.pokemons[0].name;
                console.log(msg)
                window.speechSynthesis.speak(msg);
            } else if(e.target.matches('li:nth-child(2)')) {
                msg.text = this.pokemons[1].name;
                console.log(msg)
                window.speechSynthesis.speak(msg);
            } else if(e.target.matches('li:nth-child(3)')) {
                msg.text = this.pokemons[2].name;
                console.log(msg)
                window.speechSynthesis.speak(msg);
            } else {
                msg.text = this.pokemons[3].name;
                console.log(msg)
                window.speechSynthesis.speak(msg);
            }
        }
    }
}
</script>

<style scoped>

    ul {
        display: flex;
        flex-direction: column;
        gap: 15px;
        list-style-type: none;
        padding: 0;
    }
    li {
        background-color: #fff;
        border-radius: inherit;
        border: 2px solid red;
        cursor: pointer;
        padding: 10px;
        width: 250px;
        text-transform: capitalize;
        font-weight: bold;
        color: gray;
    }

    li:hover {
        background-color: red;
        border: 2px solid red;
        color: #fff;
        transition: 1s;
    }

    .options-container {
        display: flex;
        justify-content: center;
    }

</style>