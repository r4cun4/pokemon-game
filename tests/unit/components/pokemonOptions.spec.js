import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOptions";

import { mockPokemons } from '../mocks/pokemons.mock'

describe('PokemonOptions Component', () => {

    let wrapper

    beforeEach(() => {
        wrapper = shallowMount( PokemonOptions, {
            props: {
                pokemons: mockPokemons
            }
        })
    })

    test('debe de hacer match con el snapshot', () => {

        console.log( wrapper.html() )

        expect( wrapper.html ).toMatchSnapshot()
    })

})