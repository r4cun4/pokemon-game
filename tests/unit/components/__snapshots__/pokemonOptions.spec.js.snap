// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component debe de hacer match con el snapshot 1`] = `[Function]`;
